crates/apollo-parser/src/lexer/mod.rs

```diff
@@ -55,9 +56,16 @@ impl<'a> Lexer<'a> {
             input,
             index: 0,
             finished: false,
+            // An arbitrary default limit that fits most documents without risking memory exhaustion.
+            limit: LimitTracker::new(15_000),
```

Comment:
i am thinking some of our many-mb documents will totally have more than 15K (lexical) tokens. What do you think is a good way to stress test these limits? I guess I am trying to see where to draw a line between malicious vs just very very large schemas.

Reply:
let's actually maybe talk to the router folks and see what they think about this number in particular

Reply:
yeah. 15K feels quite low to me, it's what the java impl went with, but it's probably not a good limit for a schema. it's probably fine for a query?

Reply:
i've got a few samples of queries that are totally larger than that. i ran a few that hit a recursion limit of 26K, which means they'll have even more tokens 😅 😅 😅